### PR TITLE
Fix Dijkstra shortest path algorithm + tests.

### PR DESCRIPTION
**Pull request by PatrykOlejniczak**
### Description
Connected issues #106 and suggestions from #13 .

Fix for broken unit tests and dijkstra algorithm. Simplify dijkstra code base on wikipedia pseudocode, removed unnecessary variables, little cleanup removed all Console.WriteLine etc, fixed in algorithm only suggested in #13 it was enough to get algorithm works corrects, added much more smaller unit test.

Any suggestion, improvements, more test suggestion warmly welcome.

### Checklist

- [X ] An issue was first created **before** opening this pull request
- [X] The new code follows the [contribution guidelines](CONTRIBUTING.md)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests to ensure that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Algorithms/Graphs/DijkstraShortestPaths.cs [hunk 1]
```diff
@@ -1,308 +1,167 @@
...
             }
         }
 
-        /// <summary>
-        /// Constructors helper function. Checks Optimality Conditions:
-        /// (i) for all edges e:            distTo[e.to()] <= distTo[e.from()] + e.weight()
-        /// (ii) for all edge e on the SPT: distTo[e.to()] == distTo[e.from()] + e.weight()
-        /// </summary>
-        private bool _checkOptimalityConditions(TGraph graph, TVertex source)
+        private void Initialize()
```
Comment (aalhour) [inline Algorithms/Graphs/DijkstraShortestPaths.cs]: Please use `thisCase` for private methods and `ThatCase` for public methods.

File: Algorithms/Graphs/DijkstraShortestPaths.cs [hunk 2]
```diff
@@ -1,308 +1,167 @@
...
-
-        /************************************************************************************************************/
-
-
         /// <summary>
-        /// The Dijkstra's algorithm.
+        ///     The Dijkstra's algorithm.
         /// </summary>
-        private void _dijkstra(TGraph graph, TVertex source)
+        private void Dijkstra()
```
Comment (aalhour) [inline Algorithms/Graphs/DijkstraShortestPaths.cs]: Please use `thisCase` for private methods and `ThatCase` for public methods.

File: Algorithms/Graphs/DijkstraShortestPaths.cs (hunk 1 above)
Comment (PatrykOlejniczak) [reply Algorithms/Graphs/DijkstraShortestPaths.cs]: Thanks for suggestion, fixed private naming method to be consistent with other classes.

File: Algorithms/Graphs/DijkstraShortestPaths.cs (hunk 2 above)
Comment (PatrykOlejniczak) [reply Algorithms/Graphs/DijkstraShortestPaths.cs]: Thank for suggestion. fixed :)

**Pull request merged**